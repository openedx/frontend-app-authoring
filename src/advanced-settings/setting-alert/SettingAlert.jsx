import React from 'react';
import { Alert } from '@edx/paragon';
import PropTypes from 'prop-types';

const SettingAlert = ({
  title, description, ...props
}) => (
  <Alert {...props}>
    <Alert.Heading>{title}</Alert.Heading>
    <p>{description}</p>
  </Alert>
  );

SettingAlert.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
};

SettingAlert.defaultProps = {
  title: undefined,
  description: undefined,
};

export default SettingAlert;
