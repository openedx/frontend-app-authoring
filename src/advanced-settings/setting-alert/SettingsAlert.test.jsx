import React from 'react';
import { shallow, mount } from 'enzyme';
import { Button } from '@edx/paragon';
import renderer from 'react-test-renderer';
import { CheckCircle } from '@edx/paragon/icons';
import SettingAlert from './SettingAlert';

describe('SettingAlert', () => {
  const defaultProps = {
    title: 'Test Title',
    description: 'Test Description',
    icon: CheckCircle,
  };
  it('successfully renders', () => {
    const tree = renderer.create(<SettingAlert {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('renders the title correctly in alert-heading', () => {
    const wrapper = mount(<SettingAlert {...defaultProps} />);
    const alertHeading = wrapper.find('div.alert-heading').at(0);
    expect(alertHeading.children().text()).toEqual('Test Title');
  });
  it('handles button onClick', () => {
    const mockOnClick = jest.fn();
    const wrapper = mount(
      <SettingAlert
        actions={[
          <Button onClick={mockOnClick}>Hello</Button>,
        ]}
        {...defaultProps}
      >
        Alert
      </SettingAlert>,
    );
    wrapper.find('.btn').simulate('click');
    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
  it('renders the description correctly', () => {
    const wrapper = shallow(<SettingAlert {...defaultProps} />);
    expect(wrapper.find('p').text()).toEqual('Test Description');
  });
  it('passes the additional props correctly', () => {
    const wrapper = shallow(<SettingAlert {...defaultProps} variant="success" />);
    expect(wrapper.props().variant).toEqual('success');
  });
});
