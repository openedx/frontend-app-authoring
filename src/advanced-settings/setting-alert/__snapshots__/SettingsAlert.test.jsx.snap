// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingAlert successfully renders 1`] = `
<div
  className="fade alert-content alert show"
  role="alert"
>
  <span
    className="pgn__icon alert-icon"
  >
    <svg
      aria-hidden={true}
      fill="none"
      focusable={false}
      height={24}
      role="img"
      viewBox="0 0 24 24"
      width={24}
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        fill="currentColor"
      />
    </svg>
  </span>
  <div
    className="pgn__alert-message-wrapper"
  >
    <div
      className="alert-message-content"
    >
      <div
        className="alert-heading h4"
      >
        Test Title
      </div>
      <p>
        Test Description
      </p>
    </div>
  </div>
</div>
`;
